rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -270,19 +270,41 @@ rmw_create_publisher(
     goto fail;
   }
 
+  switch (qos_policy.history) {
+    case RMW_QOS_POLICY_KEEP_LAST_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_KEEP_ALL_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
+      break;
+    default:
+      throw std::runtime_error("Unknown QoS history policy");
```

Comment:
I added a throw because of your previous comment :-)

https://github.com/ros2/rmw_connext/pull/42#discussion_r32878674

I'll replace this with an assertion.